internal/ingress/controller/parser/parser.go

```diff
@@ -1740,26 +1785,69 @@ func toKongPlugin(plugin plugin) kong.Plugin {
 	return result
 }
 
-func kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) kong.Plugin {
-	return toKongPlugin(plugin{
+func (p *Parser) kongPluginFromK8SClusterPlugin(
+	k8sPlugin configurationv1.KongClusterPlugin) (kong.Plugin, error) {
+	var configError error
+	config := k8sPlugin.Config
+	if k8sPlugin.ConfigFrom !=
+		(configurationv1.NamespacedSecretValueFromSource{}) &&
+		len(k8sPlugin.Config) > 0 {
+		return kong.Plugin{},
+			errors.Errorf("KongClusterPlugin '/%v' has both "+
+				"Config and ConfigFrom set", k8sPlugin.Name)
+	}
+	if k8sPlugin.ConfigFrom != (configurationv1.
+		NamespacedSecretValueFromSource{}) {
+		config, configError = p.namespacedSecretToConfiguration(
+			k8sPlugin.ConfigFrom)
+		if configError != nil {
+			return kong.Plugin{},
+				fmt.Errorf("error parsing config for KongClusterPlugin %v: %w",
+					k8sPlugin.Name, configError)
+		}
+	}
+	kongPlugin := toKongPlugin(plugin{
 		Name:   k8sPlugin.PluginName,
-		Config: k8sPlugin.Config,
+		Config: config,
 
 		RunOn:     k8sPlugin.RunOn,
 		Disabled:  k8sPlugin.Disabled,
 		Protocols: k8sPlugin.Protocols,
 	})
-}
-
-func kongPluginFromK8SPlugin(k8sPlugin configurationv1.KongPlugin) kong.Plugin {
-	return toKongPlugin(plugin{
+	return kongPlugin, nil
+}
+
+func (p *Parser) kongPluginFromK8SPlugin(
+	k8sPlugin configurationv1.KongPlugin) (kong.Plugin, error) {
+	var configError error
+	config := k8sPlugin.Config
+	if k8sPlugin.ConfigFrom !=
+		(configurationv1.SecretValueFromSource{}) &&
+		len(k8sPlugin.Config) > 0 {
+		return kong.Plugin{},
+			errors.Errorf("KongPlugin '%v/%v' has both "+
+				"Config and ConfigFrom set",
+				k8sPlugin.Namespace, k8sPlugin.Name)
+	}
+	if k8sPlugin.ConfigFrom != (configurationv1.SecretValueFromSource{}) {
+		config, configError = p.secretToConfiguration(
```

Comment:
Again, rename this `err`; you probabaly don't even need `configError` defined at the beginning of the function.